packages/common/src/api/tan-query/events/useCreateEvent.ts

```diff
@@ -0,0 +1,70 @@
+import { EventEventTypeEnum, EventEntityTypeEnum } from '@audius/sdk'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+
+import { useAudiusQueryContext } from '~/audius-query'
+import { Event, Feature, ID } from '~/models'
+import { toast } from '~/store/ui/toast/slice'
+
+import { getEventQueryKey } from './utils'
+
+export type CreateEventArgs = {
+  eventId?: ID
+  eventType: EventEventTypeEnum
+  userId: ID
+  entityType?: EventEntityTypeEnum
+  entityId?: ID
+  endDate?: string
+  eventData?: Record<string, any>
+}
+
+export const useCreateEvent = () => {
+  const { audiusSdk, reportToSentry } = useAudiusQueryContext()
+  const queryClient = useQueryClient()
+
+  return useMutation({
+    mutationFn: async (args: CreateEventArgs) => {
+      const sdk = await audiusSdk()
+      return await sdk.events.createEvent(args)
```

Comment:
I assume it's working if you got this far, but be careful of hashid parsing on ids if you're just passing the args through

probably sdk types would complain if it were wrong, so should be ok